apps/etterlatte-behandling/src/main/kotlin/Application.kt

```diff
@@ -1,132 +1,88 @@
+            attachContekst(dataSource())
+            sakRoutes(
+                sakService = sakService(),
+                generellBehandlingService = generellBehandlingService,
+                grunnlagsendringshendelseService = grunnlagsendringshendelseService
+            )
+            behandlingRoutes(
+                generellBehandlingService = generellBehandlingService,
+                foerstegangsbehandlingService = foerstegangsbehandlingService(),
+                revurderingService = revurderingService(),
+                manueltOpphoerService = manueltOpphoerService()
+            )
+            behandlingsstatusRoutes(behandlingsstatusService = behandlingsStatusService(), generellBehandlingService)
+            oppgaveRoutes(repo = OppgaveDao(dataSource()))
+            grunnlagsendringshendelseRoute(grunnlagsendringshendelseService = grunnlagsendringshendelseService)
         }
     }
+}
 
-    install(CallLogging) {
-        level = Level.INFO
-        filter { call -> !call.request.path().startsWith("/internal") }
-        format { call -> "<- ${call.response.status()?.value} ${call.request.httpMethod.value} ${call.request.path()}" }
-        mdc(CORRELATION_ID) { call -> call.request.header(X_CORRELATION_ID) ?: UUID.randomUUID().toString() }
-    }
-
-    install(StatusPages) {
-        exception<Throwable> { call, cause ->
-            call.application.log.error("En feil oppstod: ${cause.message}", cause)
-            call.respond(HttpStatusCode.InternalServerError, "En feil oppstod: ${cause.message}")
+class Server(private val beanFactory: BeanFactory) {
+    private val engine = embeddedServer(
+        factory = CIO,
+        environment = applicationEngineEnvironment {
+            module { module(beanFactory) }
+            connector { port = 8080 }
         }
-    }
+    )
 
-    val generellBehandlingService = beanFactory.generellBehandlingService()
-    val grunnlagsendringshendelseService = beanFactory.grunnlagsendringshendelseService()
+    fun run() = with(beanFactory) {
+        dataSource().migrate()
 
-    routing {
-        naisprobes()
-        authenticate {
-            attachContekst(dataSource)
-            sakRoutes(beanFactory.sakService(), generellBehandlingService, grunnlagsendringshendelseService)
-            behandlingRoutes(
-                generellBehandlingService,
-                beanFactory.foerstegangsbehandlingService(),
-                beanFactory.revurderingService(),
-                beanFactory.manueltOpphoerService()
-            )
-            behandlingsstatusRoutes(beanFactory.behandlingsStatusService(), generellBehandlingService)
-            route("api") {
-                oppgaveRoutes(OppgaveDao(dataSource))
-            }
-            grunnlagsendringshendelseRoute(grunnlagsendringshendelseService)
-        }
+        val behandlingHendelser = behandlingHendelser()
+        grunnlagsendringshendelseJob()
+        behandlingHendelser.start()
+
+        engine.start(true)
```

Comment:
```suggestion
        setReady().also { engine.start(true) }
```

Viss denne bruker samme readiness-mekanisme som dei andre applikasjonane, må vi kalle setReady for å seie at han er klar til å ta imot trafikk

Reply:
Spørs om ikke den ble borte i merge.. bra sett!